src/backend/src/services/teams.services.ts

```diff
@@ -73,9 +73,16 @@ export default class TeamsService {
 
     if (team.dateArchived) throw new HttpException(400, 'Cannot edit the members of an archived team');
 
-    if (team.leads.map((lead) => lead.userId).some((leadId) => userIds.includes(leadId)))
-      throw new HttpException(400, 'team leads cannot be members!');
-
+    // if the new members array includes a current lead on that team, that member will be deleted as a lead of that team
+    const newLeadsArr: number[] = [];
```

Comment:
look into replacing all of this logic with the filter function, should be able to do 
`newLeadsArr = team.lead.filter((lead) => {some logic here about how userIds doesn't include lead.userId})`